pandas/tools/plotting.py

```diff
@@ -1217,8 +1217,22 @@ def _adorn_subplots(self):
 
         if self.title:
             if self.subplots:
-                self.fig.suptitle(self.title)
+                if is_list_like(self.title):
+                    if len(self.title) != len(self.axes):
+                        msg = 'The length of `title` must equal the number ' \
```

Comment:
It'd be nice to report the length of `title` and the expected number of columns in the error message.